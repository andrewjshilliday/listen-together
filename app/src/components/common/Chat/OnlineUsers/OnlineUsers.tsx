import React from 'react';
import { useSelector } from 'react-redux';
import { Avatar, Tooltip  } from 'antd';
import { ApplicationState } from '../../../../store';
import styled from 'styled-components';

const OnlineUsers: React.FC = (props: any) => {
  const users = useSelector((state: ApplicationState) => state.room.users);

  return (
    <OnlineUsersContainer>
      {
        users.length > 0 ?
          <>
            <span>Other people in this room</span>
            <UserAvatarContainer>
              {
                users.map(user =>
                  <Tooltip key={user.id} title={user.name}>
                    <AvatarIcon size="large" color={user.color}>{user.name}</AvatarIcon>
                  </Tooltip>)
              }
            </UserAvatarContainer>
          </>
          :
          <span>Looks like you're on your own here!</span>
      }
    </OnlineUsersContainer>
  );
}

export default OnlineUsers;


const OnlineUsersContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-height: 75px;
  margin: 0 25px;
  border-bottom: 1px solid #ccc;
`;
const UserAvatarContainer = styled.div`
  margin: 6px 0;
`;
const AvatarIcon = styled(Avatar)<{color: string}>`
  margin: 0 5px;
  background-color: ${props => props.color};
`;
