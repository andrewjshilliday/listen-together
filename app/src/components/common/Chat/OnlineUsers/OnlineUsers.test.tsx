import React from 'react';
import renderer from 'react-test-renderer';
import OnlineUsers from './OnlineUsers';

jest.mock('../../../providers/RoomProvider', () => require('../../../../__mocks__/providers/RoomProvider.mock').default);

it('renders correctly', () => {
  const tree = renderer.create(<OnlineUsers />).toJSON();
  expect(tree).toMatchSnapshot();
});



/* import React from 'react';
import renderer from 'react-test-renderer';
import OnlineUsers from './OnlineUsers';

const spyReturns = (returnValue: any) => jest.fn(() => returnValue);

describe('scenario', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  const setup = (users: string[] = [], mockOverrides?: any) => {
    const mockedRoomProvider = {
      useRoom: () => ({
        users: users.map((user, index) => ({ name: user, id: index, color: '#cfcfcf' })),
        // b: spyReturns(true),
        ...mockOverrides
      })
    }
    return {
      mockedModule: jest.doMock('../../../providers/RoomProvider', () => mockedRoomProvider)
    }
  }

  it('renders correctly', () => {
    const { mockedModule } = setup();
    const tree = renderer.create(<OnlineUsers />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple users', () => {
    const { mockedModule } = setup(['user1', 'user2']);
    const tree = renderer.create(<OnlineUsers />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}); */
