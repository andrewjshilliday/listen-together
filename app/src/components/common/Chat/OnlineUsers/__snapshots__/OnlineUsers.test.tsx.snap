// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm fuPtXB"
>
  <span>
    Other people in this room
  </span>
  <div
    className="sc-AxirZ eLtbJv"
  >
    <span
      className="ant-avatar sc-AxiKw hiMjsU ant-avatar-lg ant-avatar-circle"
      color="#cfcfcf"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <span
        className="ant-avatar-string"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        user0
      </span>
    </span>
    <span
      className="ant-avatar sc-AxiKw hiMjsU ant-avatar-lg ant-avatar-circle"
      color="#cfcfcf"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <span
        className="ant-avatar-string"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        user1
      </span>
    </span>
  </div>
</div>
`;
